acceptance/github_asset_fetcher.go

```diff
@@ -0,0 +1,420 @@
+package acceptance
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/Masterminds/semver"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/google/go-github/v30/github"
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/pack/internal/blob"
+	"github.com/buildpacks/pack/internal/config"
+	ilogging "github.com/buildpacks/pack/internal/logging"
+)
+
+const (
+	assetCacheDir      = "test-assets-cache"
+	assetCacheManifest = "github.json"
+)
+
+type GithubAssetFetcher struct {
+	ctx          context.Context
+	logger       *ilogging.LogWithWriters
+	githubClient *github.Client
+	cacheDir     string
+}
+
+type assetCache map[string]map[string]cachedRepo
+type cachedRepo struct {
+	Assets   cachedAssets
+	Sources  cachedSources
+	Versions cachedVersions
+}
+type cachedAssets map[string][]string
+type cachedSources map[string]string
+type cachedVersions map[string]string
+
+func NewGithubAssetFetcher() (*GithubAssetFetcher, error) {
+	packHome, err := config.PackHome()
+	if err != nil {
+		return nil, errors.Wrap(err, "getting pack home")
+	}
+
+	return &GithubAssetFetcher{
+		ctx:          context.TODO(),
+		logger:       ilogging.NewLogWithWriters(os.Stdout, os.Stderr), // TODO: not sure if this is what we really want
```

Comment:
Logging with testing is a little bit awkward. I'm using `fmt.Printf` in places to get the output to show while the test is running. Otherwise it would print when the test finishes and only if there's a failure. I'm not sure what we want.